models/esss.R

```diff
@@ -0,0 +1,111 @@
+                               estimate = ets_forecast$mean,
+                               LowerPI = ets_forecast$lower[ , CI_match], 
+                               UpperPI = ets_forecast$upper[ , CI_match],
+                               fit_start_newmoon = fit_start_newmoon,
+                               fit_end_newmoon = fit_end_newmoon,
+                               initial_newmoon = initial_newmoon)
+    for(i in 5:8)
+      output_fcast[ , i] <- as.character(output_fcast[ , i])
+    for(i in 9:11)
+      output_fcast[ , i] <- as.numeric(output_fcast[ , i])
+
+    output_aic <- data.frame(date = as.Date(forecast_date), 
+                             currency = 'abundance', 
+                             model = 'ESSS', 
+                             level = level, species = 'total', 
+                             aic = as.numeric(ets_model$aic), 
+                             fit_start_newmoon = fit_start_newmoon,
+                             fit_end_newmoon = fit_end_newmoon,
+                             initial_newmoon = initial_newmoon)
+    for(i in 2:5)
+      output_aic[ , i] <- as.character(output_aic[ , i])
+
+    output <- list(output_fcast, output_aic)
+    names(output) <- c("forecast", "aic")
+
+    return(output)
+  }
+
+  all <- read.csv("data/rodent_all.csv")
+  controls <- read.csv("data/rodent_controls.csv")
+  model_metadata <- yaml.load_file("data/model_metadata.yaml")
+  forecast_date <- as.Date(model_metadata$forecast_date)
+  filename_suffix <- model_metadata$filename_suffix
+  forecast_months <- model_metadata$forecast_months
+  forecast_years <- model_metadata$forecast_years
+  forecast_newmoons <- model_metadata$forecast_newmoons
+
+  forecasts_all <- forecast_esss(abundances = all, 
+                                 forecast_date = forecast_date,
+                                 forecast_months = forecast_months, 
+                                 forecast_years = forecast_years,
+                                 forecast_newmoons = forecast_newmoons,
+                                 level = "All",
+                                 num_forecast_newmoons = 12, 
+                                 CI_level = 0.9)
+
+  forecasts_controls <- forecast_esss(abundances = controls, 
+                                      forecast_date = forecast_date,
+                                      forecast_months = forecast_months, 
+                                      forecast_years = forecast_years,
+                                      forecast_newmoons = forecast_newmoons,
+                                      level = "Controls",
+                                      num_forecast_newmoons = 12, 
+                                      CI_level = 0.9)
+
+  forecasts <- rbind(forecasts_all[[1]], forecasts_controls[[1]])
+  aics <- rbind(forecasts_all[[2]], forecasts_controls[[2]])
+
+  fcast_path <- paste("ESSS", filename_suffix, ".csv", sep = "")
+  fcast_path <- file.path('tmp', fcast_path)
```

Comment:
I'd probably put these all on one line (shortening `filename_suffix` if needed for length):

```
fcast_path <- file.path('tmp', paste("ESSS", filename_suffix, ".csv", sep = ""))
```

If we want to keep them on separate lines the first name should be descriptive `fcast_filename`.

Reply:
the issue with shortening to get them all on one line is that the aic file's path name generation is longer (due to the addition of "_model_aic.csv")
as a result `filename_suffix` would have to be shortened to 3 characters, given everything else

given that, I'm going to go with the more descriptive object names